src/EnergyPlus/ExtendedHeatIndex.cc

```diff
@@ -0,0 +1,552 @@
+            [&](Real64 Tf) { return (Tf - Ta) / Ra_bar(Tf, Ta) + (Pc - Pa) / (ZsRs + Za_bar) - (Tc - Tf) / Rs; },
+            std::max(0.0, std::min(Tc, Ta) - Rs * std::abs(m_bar)),
+            std::max(Tc, Ta) + Rs * std::abs(m_bar));
+        Real64 const flux1 = Q - Qv(Ta, Pa) - (1.0 - phi) * (Tc - Ts) / Rs;
+        Real64 const flux2 = Q - Qv(Ta, Pa) - (1.0 - phi) * (Tc - Ts) / Rs - phi * (Tc - Tf) / Rs;
+        Real64 Rf;
+
+        if (flux1 <= 0.0) {
+            varname = static_cast<int>(EqvarName::Phi);
+            phi = 1.0 - (Q - Qv(Ta, Pa)) * Rs / (Tc - Ts);
+            return phi;
+        } else if (flux2 <= 0.0) {
+            varname = static_cast<int>(EqvarName::Rf);
+            Real64 const Ts_bar = Tc - (Q - Qv(Ta, Pa)) * Rs / phi + (1.0 / phi - 1.0) * (Tc - Ts);
+            General::SolveRoot(
+                state,
+                tol,
+                maxIter,
+                SolFla,
+                Tf,
+                [&](Real64 Tf) {
+                    return (Tf - Ta) / Ra_bar(Tf, Ta) + (Pc - Pa) * (Tf - Ta) / ((ZsRs + Za_bar) * (Tf - Ta) + r * Ra_bar(Tf, Ta) * (Ts_bar - Tf)) -
+                           (Tc - Ts_bar) / Rs;
+                },
+                Ta,
+                Ts_bar);
+            Rf = Ra_bar(Tf, Ta) * (Ts_bar - Tf) / (Tf - Ta);
+            return Rf;
+        } else {
+            Real64 const flux3 = Q - Qv(Ta, Pa) - (Tc - Ta) / Ra_un(Tc, Ta) - (phi_salt * pvstar(Tc) - Pa) / Za_un;
+            if (flux3 < 0.0) {
+                varname = static_cast<int>(EqvarName::Rs);
+                General::SolveRoot(
+                    state,
+                    tol,
+                    maxIter,
+                    SolFla,
+                    Ts,
+                    [&](Real64 Ts) { return (Ts - Ta) / Ra_un(Ts, Ta) + (Pc - Pa) / (Zs((Tc - Ts) / (Q - Qv(Ta, Pa))) + Za_un) - (Q - Qv(Ta, Pa)); },
+                    0.0,
+                    Tc);
+                Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                ZsRs = Zs(Rs);
+                Real64 const Ps = Pc - (Pc - Pa) * ZsRs / (ZsRs + Za_un);
+                if (Ps > phi_salt * pvstar(Ts)) {
+                    General::SolveRoot(
+                        state,
+                        tol,
+                        maxIter,
+                        SolFla,
+                        Ts,
+                        [&](Real64 Ts) { return (Ts - Ta) / Ra_un(Ts, Ta) + (phi_salt * pvstar(Ts) - Pa) / Za_un - (Q - Qv(Ta, Pa)); },
+                        0.0,
+                        Tc);
+                    Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                }
+                return Rs;
+            } else {
+                varname = static_cast<int>(EqvarName::DTcdt);
+                Rs = 0.0;
+                dTcdt = (1.0 / C) * flux3;
+                return dTcdt;
+            }
+        }
+    }
+
+    // Convert the find_T function
+    Real64 find_T(EnergyPlusData &state, int const eqvar_name, Real64 const eqvar)
```

Comment:
You can use `enum`s as parameters, why make it an `int` here?

Reply:
yeah also need to use enum class directly here as well